networking/src/main/java/com/lyft/networking/LyftApiFactory.java

```diff
@@ -21,20 +22,67 @@ public LyftApiFactory(ApiConfig apiConfig) {
     /**
      * @return An implementation of Lyft's Public API endpoints that do not require a user.
      * The return type of API calls will be {@link retrofit2.Call}. Used by the LyftButton.
-     * See: <a href="http://petstore.swagger.io/?url=https://api.lyft.com/v1/spec#!/Public/">http://petstore.swagger.io/?url=https://api.lyft.com/v1/spec#!/Public/</a>
+     *
+     * By default, the Retrofit client will throw a {@link com.lyft.networking.exceptions.PartialResponseException}
+     * if the response returned by the server has missing information (i.e. non-null values returned as null).
+     *
+     * THE CALLER MUST be able to handle {@link com.lyft.networking.exceptions.PartialResponseException}.
+     * Unhandled exceptions will cause a runtime crash.
+     *
+     * To explicitly disallow this, invoke #getLyftPublicApi(boolean).
      */
     public LyftPublicApi getLyftPublicApi() {
-        Retrofit retrofitPublicApi = getRetrofitBuilder(getPublicOkHttpClient()).build();
+        return getLyftPublicApi(true);
+    }
+
+    /**
+     * Creates an implementation of Lyft's Public API, with an option of generating {@link com.lyft.networking.exceptions.PartialResponseException}
+     * should the server return incomplete response models.
+     *
+     * @param requiresCompleteResponse A boolean flag to indicate whether an exception should be thrown
+     *                                 if the server returns partial data. If set true,
+     *                                 the Retrofit client will throw a {@link com.lyft.networking.exceptions.PartialResponseException}
+     *                                 when the response returned by the server has missing information (i.e. non-null values returned as null).
+     *
+     *                                 ONLY SET AS FALSE if the consumer is responsible for manually handling null checks.
+     *
+     * @return An implementation of Lyft's Public API endpoints that do not require a user.
+     * The return type of API calls will be {@link retrofit2.Call}. Used by the LyftButton.
+     */
+    public LyftPublicApi getLyftPublicApi(boolean requiresCompleteResponse) {
```

Comment:
This is a boolean trap: https://ariya.io/2011/08/hall-of-api-shame-boolean-trap
Don't be John Carmack worst enemy https://twitter.com/id_aa_carmack/status/367485612627984385?lang=en
:)

What about renaming this `getUnsafeLyftPublicApi()` with no parameter to make it clear that this will need extract work from the user?
This is valid for other methods in this file.